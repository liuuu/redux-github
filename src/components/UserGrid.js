import React from "react";
import { Col, Card, Avatar } from "antd";
import { Link } from "react-router-dom";

const UserGrid = props => {
  return (
    <Col span={8} style={(props.style, { padding: "20px" })}>
      <Link to={`/user/${props.owner.login}`} style={{ color: "black" }}>
        <Card>
          <div className="popular-rank">
            #{parseInt(`${props.index}`, 10) + 1}
          </div>
          <ul className="space-list-item">
            <li>
              <Avatar
                className="avatar"
                src={props.owner.avatar_url}
                alt={props.owner.login}
              />
            </li>
            <li>
              <p>
                {props.name}
                {props.id}
              </p>
            </li>
            <li>
              @{props.owner.login}
            </li>
            <li>
              Stars: {props.stargazers_count}
            </li>
          </ul>
        </Card>
      </Link>
    </Col>
  );
};

export default UserGrid;
