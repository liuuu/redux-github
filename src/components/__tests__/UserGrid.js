import React from "react";
import UserGrid from "../UserGrid";
import { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

const props = {
  style: "any",
  owner: {
    login: "any"
  }
};
it("should be good", () => {
  const wrapper = shallow(<UserGrid {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
