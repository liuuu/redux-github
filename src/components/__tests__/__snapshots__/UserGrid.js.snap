// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be good 1`] = `
<Col
  span={8}
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Link
    replace={false}
    style={
      Object {
        "color": "black",
      }
    }
    to="/user/any"
  >
    <Component>
      <div
        className="popular-rank"
      >
        #
        NaN
      </div>
      <ul
        className="space-list-item"
      >
        <li>
          <Avatar
            alt="any"
            className="avatar"
            prefixCls="ant-avatar"
            shape="circle"
            size="default"
          />
        </li>
        <li>
          <p />
        </li>
        <li>
          @
          any
        </li>
        <li>
          Stars: 
        </li>
      </ul>
    </Component>
  </Link>
</Col>
`;
